### ISSUE TITLE
setInterval every switch without clearing when autoPlay

### ISSUE THREAD

**Issue by afc163** [akiran/react-slick#285]
Cause issue after react-slick is ummounted.

```
warning.js?0260:44 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the InnerSlider component.
```

version: 0.11.1

https://github.com/ant-design/ant-design/issues/1397#issuecomment-209340907

---

https://github.com/akiran/react-slick/commit/9bbe0fa9191bb5b1156cec4755532ee108f14cbc

Why use `setInterval` instead `setTimeout`, `setInterval` will repeat forever, and never clear it when it switch automaticly!! And `componentWillUnmount` can only clear one last interval (there are many intervals).

**Comment by likethemammal** [comment]
:+1: Ran into this too

**Comment by Nierrrrrrr** [comment]
Ran into this, too. And manually fixed by edit the src file as pull #287.

**Comment by kairi1227** [comment]
please fix  ~~~~
